Neos.Neos/Classes/Controller/Module/Administration/UsersController.php

```diff
@@ -317,4 +375,32 @@ protected function getAuthenticationProviders()
         sort($providerNames);
         return array_combine($providerNames, $providerNames);
     }
+
+    /**
+     * Returns the roles that the current editor is able to assign
+     * Administrator can assign any roles, other users can only assign their own roles
+     *
+     * @return array
+     */
+    protected function getAllowedRoles(): array
+    {
+        $currentUserRoles = $this->userService->getAllRoles($this->currentUser);
+        return $this->isAdministrator ? $this->policyService->getRoles() : $currentUserRoles;
+    }
+
+    /**
+     * Returns whether the current user is allowed to edit the given user.
+     * Administrators can edit anybody.
+     *
+     * @param User $user
+     */
+    protected function isEditingAllowed(User $user): bool
+    {
+        if ($this->isAdministrator) {
+            return true;
+        }
+        $currentUserRoles = $this->userService->getAllRoles($this->currentUser);
+        $userRoles = $this->userService->getAllRoles($user);
+        return count(array_diff($userRoles, $currentUserRoles)) === 0;
```

Comment:
That would mean, editing is only allowed when the user to edit has the same roles than me. That makes no sense to me.